grails-app/assets/javascripts/digivol-notebook.js

```diff
@@ -69,10 +69,16 @@ var notebook = {
      * @param id
      */
     function load_content(marker, id) {
+      const taskViewUrl = $('#map').attr('taskview-url');
       $.ajax($('#map').attr('infowindow-url') + "/" + id).done(function(data) {
         var content =
-          "<div style='font-size:12px;line-height:1.3em;'>Catalogue No.: " + data.cat + "<br/>Taxon: " + data.name + "<br/>Transcribed by: " + data.transcriber +
-          "</div>";
+            "<div style='font-size:12px;line-height:1.3em;'>" +
+            "Task: " + id +
+            "<br />File: ";
+        if (taskViewUrl !== "" && taskViewUrl !== undefined) content += "<a href=\"" + taskViewUrl + "/" + id + "\" target=\"_blank\">" + data.filename + "</a>";
```

Comment:
Just a fussy comment, but you've got two different "does this exist?" tests here.  A truthy check would suffice probably but I'm not up to date on current javascript best practice...

Reply:
You are correct, I'm just paranoid about undefined edge cases. I've removed the extra check.